### PR TITLE
ciao-cert: hostname clarifications for README.md

### PR DESCRIPTION
**Pull request by tpepper**
We're seeing users not understanding the implications of hostnames in
certificates.  This commit attempts to be more explicit about the security
trade offs and the deployment choices for generic keys versus machine
specific keys.

Signed-off-by: Tim Pepper <timothy.c.pepper@linux.intel.com>

### CODE REVIEW

File: ciao-cert/README.md [hunk 1]
```diff
@@ -51,63 +51,66 @@ Usage of ciao-cert:
...
-We need to generate 5 private keys for the Scheduler, Controller, Networking Agent,
-CNCI Agent and the Compute Node Agents. We also need to generate the CA certificate.
+At a minimum, need to generate 5 private keys for the Scheduler,
+Controller, Networking Agent, CNCI Agent and the Compute Node Agents. We
+also need to generate the CA certificate.
 
-* Scheduler private key and CA certificate
+Depending on your security tolerances, you may choose one of two deployment
+styles for SSNTP client certificates:
+* a generic private key for each SSNTP client type: ```-host=localhost``` in the command examples below.  Each Compute Node agent machine would have a copy of the one compute node agent key with 'localhost' validity.  This may be easiest for a fleet of ephemeral test systems.
```
Comment (tpepper) [inline ciao-cert/README.md]: But what do I change it to in the example?  Does each runtime instance of the CNCI workload's base image have a unique cert in it at launch time?

Comment (mcastelino) [inline ciao-cert/README.md]: @tpepper the CNCI cert is not an issue. The scheduler is the only one that has to have a FQDN cert that resolves. As the CNCI agent and in fact even the launchers use to connect back to the scheduler.

So localhost should not be used for the scheduler cert.

Comment (tpepper) [inline ciao-cert/README.md]: Ah I see what you're saying.  I'll clarify that the scheduler MUST have a fqdn specified.

Comment (tpepper) [inline ciao-cert/README.md]: Slightly revised text pushed.

**Comment by mcastelino** [comment]
LGTM

File: ciao-cert/README.md [hunk 2]
```diff
@@ -51,63 +51,68 @@ Usage of ciao-cert:
...
 
 * Scheduler private key and CA certificate
 
-```shell
-$GOBIN/ciao-cert -server -role scheduler -email=ciao-devel@lists.clearlinux.org -organization=Intel -ip=192.168.1.118 -host=ciao-ctl.intel.com -verify
-```
-
-That will generate `CAcert-ciao-ctl.intel.com.pem` and `cert-Scheduler-ciao.ctl.intel.com.pem`.
+        $GOBIN/ciao-cert -server -role scheduler -email=ciao-devel@lists.clearlinux.org -organization=Intel -ip=192.168.1.118 -host=ciao-ctl.intel.com -verify
+  That will generate `CAcert-ciao-ctl.intel.com.pem` and `cert-Scheduler-ciao.ctl.intel.com.pem`.
```
Comment (markdryan) [inline ciao-cert/README.md]: @tpepper One quick question.  Why do we need to specify the IP address and the hostname when creating a scheduler certificate?

**Comment by rbradford** [comment]
FYI, the unit tests *require* certs generated for localhost (at least for the controller tests.)

File: ciao-cert/README.md (hunk 2 above)
Comment (tpepper) [inline ciao-cert/README.md]: No idea.  That line's a white space change.  @sameo ?

**Comment by tpepper** [comment]
@rbradford do you think the cert README should mention that unit tests need localhost certs?  Or the unit tests' docu?

**Comment by markdryan** [comment]
I'm just going to merge this.  The specification of the ipaddress in scheduler is not added by this PR.  It's always been there.  I'll clarify with @sameo as to whether it's really needed.

**Pull request merged**